src/analysis/inner_product_space/pi_L2.lean

```diff
@@ -381,3 +381,88 @@ def linear_isometry_equiv.from_orthogonal_span_singleton
   (n : ℕ) [fact (finrank 𝕜 E = n + 1)] {v : E} (hv : v ≠ 0) :
   (𝕜 ∙ v)ᗮ ≃ₗᵢ[𝕜] (euclidean_space 𝕜 (fin n)) :=
 linear_isometry_equiv.of_inner_product_space (finrank_orthogonal_span_singleton hv)
+
+section complex
+
+variables {V : Type*}
+[inner_product_space 𝕜 V]
+[finite_dimensional 𝕜 V]
+
+variables {S : submodule 𝕜 V} {L : S →ₗᵢ[𝕜] V}
+
+instance S_complete : complete_space S := finite_dimensional.complete 𝕜 S
+
+instance V_complete : complete_space V := finite_dimensional.complete 𝕜 V
```

Comment:
Typically you'd instantiate these inside the proof that uses them, like
```lean
  haveI : complete_space S := finite_dimensional.complete 𝕜 S,
```

Reply:
`haveI` is new to me, and I think I've wanted it before.  Thanks!